src/pydata_sphinx_theme/__init__.py

```diff
@@ -140,13 +140,15 @@ def update_templates(app, pagename, templatename, context, doctree):
     # Add links for favicons in the topbar
     for favicon in context.get("theme_favicons", []):
         icon_type = Path(favicon["href"]).suffix.strip(".")
+        opts = {
+            "rel": favicon.get("rel", "icon"),
+            "sizes": favicon.get("sizes", "16x16"),
+            "type": f"image/{icon_type}",
+        }
+        if favicon.get("color"):
```

Comment:
```suggestion
        if "color" in favicon:
```

get will raise an exception if no color is set